schemas/input/csl-citation.json

```diff
@@ -70,86 +70,114 @@
-          "type": "string"
-        },
-        "volume": {
-          "type": "string"
+    "locator-variable": {
+      "description": "A locator is a list, that can be a string, or a (preferred) array.",
+      "anyOf": [
+        {
+          "description": "When using the string ...",
+          "type": "string"
+        },
+        {
+          "description": "Strings represent a literal number, while numbers (integer or floating point) should otherwise be used. Ranges are represented by a two-item inner array.",
+          "examples": [
+            { "page": [2, 3, [5, 12]] },
+            { "page": ["I-13"] },
+            { "page": [["I-13", "I-18"]] }
+          ],
+          "anyOf": [
+            { "type": "string" },
+            { "type": "number" },
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/locators"
+              }
+            }
+          ]
         }
+      ]
+    },
+    "locators": {
+      "appendix": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "article": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "book": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "canon": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "chapter": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "column": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "elocation": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "equation": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "figure": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "folio": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "issue": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "line": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "note": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "opus": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "page": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "paragraph": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "part": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "rule": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "section": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "sub-verbo": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "supplement": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "table": {
+        "$ref": "#/definitions/locator-variable"
+      },
+      "timestamp": {
+        "description": "An hour, minute, second timestamp for video or audio sources.",
+        "type": "string",
+        "pattern": "^[0-9]{2}:[0-9]{2}(:[0-9]{2})?$",
+        "examples": ["01:12:32"]
+      },
+      "title": {
```

Comment:
Two technical issues ATM:

1. This is causing a conflict, and an error, because JSON schema has a reserved `title` property for annotations. If I move it out of the definitions area, it goes away, but then I don't know how to refer to it so it can be used within the array.

2. I haven't figured out how to remove the `locators` array.

Reply:
"title" as in of a law ala "Civil Rights Act of 1964, Title VII". 

We could change it to "title-locator".